File: py/src/practise150/p10_valid_palindrome_test.py
from src.practise150.p10_valid_palindrome import Solution

def test_valid_palindrome():
    test_cases=[
        ("Was it a car or a cat I saw?", True),
        ("tab a cat", False)
    ]
    
    s= Solution()
    for test_case in test_cases:
        result = s.valid_palindrome(test_case[0])
        assert result == test_case[1]

File: go/practise150/p10_valid_palindrome_test.go
package practise150

import (
	"fmt"
	"testing"
)

func TestValidPalindrome(t *testing.T) {
	test_cases := []struct {
		input    string
		expected bool
	}{
		{
			input:    "Was it a car or a cat I saw?",
			expected: true,
		},
		{
			input:    "tab a cat",
			expected: false,
		},
	}

	for idx, test_case := range test_cases {
		t.Run(fmt.Sprintf("Test_case_no:%d", idx+1), func(t *testing.T) {
			actual := isPalindrome(test_case.input)
			if actual != test_case.expected {
				t.Errorf("Testing isPalindrome function. Input: %s, Expected: %v, Actual: %v", test_case.input, test_case.expected, actual)
			}
		})
	}

}
